caddyhttp/proxy/policy.go

```diff
@@ -100,3 +103,30 @@ func (r *RoundRobin) Select(pool HostPool) *UpstreamHost {
 	}
 	return nil
 }
+
+// IPHash is a policy that selects hosts based on hashing the request ip
+type IPHash struct{}
+
+func hash(s string) uint32 {
+	h := fnv.New32a()
+	h.Write([]byte(s))
+	return h.Sum32()
+}
+
+// Select selects an up host from the pool using a round robin ordering scheme.
+func (r *IPHash) Select(pool HostPool, request *http.Request) *UpstreamHost {
+	poolLen := uint32(len(pool))
+	hash := hash(request.RemoteAddr)
```

Comment:
You may be right. I would recommend `net.SplitHostPort()` instead, however. IPv6 addresses would otherwise be problematic.